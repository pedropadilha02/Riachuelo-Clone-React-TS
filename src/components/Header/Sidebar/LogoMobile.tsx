import styled from 'styled-components';
import { useState } from 'react';

const Logo = styled.h1`
  display: block;

  @media (max-width: 1200px) {
    opacity: 1;
    height: 60px;
    line-height: 60px;
    text-align: center;
    margin: 0;
    position: absolute;
    width: 100%;
  }
`;

export function LogoMobile() {
  return (
    <Logo>
      <img src="http://localhost:3000/assets/rchlo.svg" alt="" />
    </Logo>
  );
}
