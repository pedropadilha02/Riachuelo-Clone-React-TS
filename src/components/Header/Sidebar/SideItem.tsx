import styled from 'styled-components';

const MenuItemStyled = styled.li`
  list-style: none;

  &:hover {
    border: none;
    background-color: darkgrey;
  }

  @media (max-width: 1200px) {
    width: 100%;
    padding-top: 10px;
    padding-left: 15px;
    padding-bottom: 10px;
    padding-right: 0px;
    border-bottom: 1px solid gray;
  }
`;

const MenuLink = styled.a`
  width: 100%;
  hegiht: 100%;
  font-size: 0.8125rem;
  font-weight: 600;
  color: #000;
  text-transform: uppercase;
  padding-bottom: 5px;
  vertical-align: baseline;
  white-space: nowrap;
`;

interface CategoriaData {
  categoria: { id: string; label: string };
}

export function SideItem({ categoria }: CategoriaData) {
  return (
    <MenuItemStyled>
      <MenuLink>{categoria.label}</MenuLink>
    </MenuItemStyled>
  );
}
