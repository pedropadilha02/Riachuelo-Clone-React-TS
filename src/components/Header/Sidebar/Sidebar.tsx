import styled from 'styled-components';
import { useState, useEffect } from 'react';
import { LogoMobile } from './LogoMobile';
import { SideItem } from './SideItem';

const MenuMobile = styled.div`
  display: none;
  margin-bottom: 0px;

  @media (max-width: 1200px) {
    display: flex;
    align-items: center;
  }
`;

const MenuHeader = styled.div`
  height: 70px;
  width: 100%;
`;

const MenuButton = styled.div`
  margin-top: 20px;
  margin-left: 20px;
  padding: 10px;
  width: 28px;
  height: 28px;
  opacity: 54%;
  display: flex;
  flex-direction: column;
  align-items: center;
`;

const MenuIcon = styled.img`
  width: 24px;
  padding-bottom: 3px;
  margin: 0;
  border-radius: 50%;
  &:active {
    background-color: grey;
  }
`;

const MenuTitle = styled.p`
  margin: 0;
  text-transform: lowercase;
  font-size: 0.5625rem;
  font-weight: 400;
  line-height: 0;
`;

const MenuList = styled.ul`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  padding-left: 0px;
  margin: 10px 0px;
  margin-right: 0px;
  width: 80%;
  flex-wrap: nowrap;
  z-index: 9999;
`;

interface CategoriaData {
  id: string;
  label: string;
}

let ChangeMenu: string = '';
let ChangeTitle: string = '';
let openMenu: string = '';

export function Sidebar() {
  const [categories, setCategories] = useState<CategoriaData[]>([]);
  const [menuOpen, setMenuOpen] = useState(false);

  useEffect(() => {
    fetch('http://localhost:5000/categories')
      .then((res) => res.json())
      .then((res) => setCategories(res));
  }, []);

  if (!menuOpen) {
    ChangeMenu = 'http://localhost:3000/assets/menu.svg';
    ChangeTitle = 'menu';
    openMenu = '-100%';
  } else {
    ChangeMenu = 'http://localhost:3000/assets/close.svg';
    ChangeTitle = 'fechar';
    openMenu = '0%';
  }

  const MenuStyled = styled.nav`
    opacity: 0;

    @media (max-width: 1200px) {
      opacity: 1;
      position: absolute;
      width: 80%;
      top: 100px;
      background-color: white;
      z-index: 501;
      left: ${openMenu};
    }
  `;

  return (
    <MenuMobile>
      <MenuHeader>
        <LogoMobile />
        <MenuButton onClick={() => setMenuOpen(!menuOpen)}>
          <MenuIcon src={ChangeMenu} />
          <MenuTitle>{ChangeTitle}</MenuTitle>
        </MenuButton>
      </MenuHeader>

      <MenuStyled>
        <MenuList>
          {categories?.map((categoria) => (
            <SideItem key={categoria.id} categoria={categoria} />
          ))}
        </MenuList>
      </MenuStyled>
    </MenuMobile>
  );
}
