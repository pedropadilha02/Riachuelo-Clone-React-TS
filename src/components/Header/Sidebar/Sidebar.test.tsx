import { render } from '@testing-library/react';
import { Sidebar } from './Sidebar';

describe('Teste de renderização do CopyRight', () => {
  it('O snapshot do component deve permanecer semre o mesmo', () => {
    const { container } = render(<Sidebar />);

    expect(container).toMatchSnapshot();
  });
});
