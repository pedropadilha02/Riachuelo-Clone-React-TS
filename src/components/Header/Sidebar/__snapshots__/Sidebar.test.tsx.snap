// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teste de renderização do CopyRight O snapshot do component deve permanecer semre o mesmo 1`] = `
<div>
  <div
    class="sc-hKFxyN ckjeHz"
  >
    <div
      class="sc-eCApnc exFulq"
    >
      <h1
        class="sc-dlnjwi lvQDz"
      >
        <img
          alt=""
          src="http://localhost:3000/assets/rchlo.svg"
        />
      </h1>
      <div
        class="sc-jSFjdj iAfa-Dv"
      >
        <img
          class="sc-gKAaRy hfPqRA"
          src="assets/menu.svg"
        />
        <p
          class="sc-iCoGMd eceaLO"
        >
          menu
        </p>
      </div>
    </div>
    <nav
      class="sc-pNWdM InFUy"
    >
      <ul
        class="sc-fujyAs QkLnU"
      />
    </nav>
  </div>
</div>
`;
